packages/core/src/index.ts

```diff
@@ -256,46 +254,74 @@ export const clearAllUserPersonas = (): Promise<boolean> => {
   return EmbraceManagerModule.clearAllUserPersonas();
 };
 
-export const logMessage = (
+const logMessage = (
   message: string,
-  severity: "info" | "warning" | "error" = "error",
-  properties?: Properties,
+  severity:
+    | LogSeverity.INFO
+    | LogSeverity.WARNING
+    | LogSeverity.ERROR = LogSeverity.ERROR,
+  // Android Native method is handling the null case
+  // iOS Native method is waiting for a non-nullable object
+  properties: LogProperties = {},
+  includeStacktrace?: boolean,
 ): Promise<boolean> => {
-  const stacktrace = severity === INFO ? "" : generateStackTrace();
+  const stackTrace =
+    // `LogSeverity.INFO` are not supposed to send stack traces
+    // this is also restricted in the Native layers
+    includeStacktrace && severity !== LogSeverity.INFO
+      ? generateStackTrace()
+      : "";
+
+  if (properties === null) {
+    console.warn(
+      "[Embrace] `properties` is null. It should be an object of type `Properties`. Native layer will ignore this value.",
```

Comment:
should we clarify that it could be also be `undefined`?

Reply:
if `properties` is passed specifically as `undefined` then it will take the default value, that is an empty object `{}` so it won't take any effect

Reply:
ah right